components/EditCoverSection.tsx

```diff
@@ -1,9 +1,271 @@
+
+  const [imageLink, setImageLink] = useState<string>();
+
+  const [isOpen, setIsOpen] = useState<string | null>(null);
+
+  useEffect(() => {
+    fetchHome().then((res: Home | null) => {
+      setHome(res || {});
+      setDebouncedHome(res || {});
+    });
+  }, []);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setHome(debouncedHome);
+    }, 1000);
+
+    return () => clearTimeout(timerId);
+  }, [debouncedHome]);
+
+  useEffect(() => {
+    // updateHome(home);
+  }, [home]);
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { name, value } = e.target;
+    setDebouncedHome({ ...home, [name]: value });
+  };
+
+  const handleOpen = (name: string): void => {
+    switch (name) {
+      case "cover":
+        setIsOpen(isOpen === name ? null : name);
+        break;
+      case "cover2":
+        setIsOpen(isOpen === name ? null : name);
+        break;
+      case "cover3":
+        setIsOpen(isOpen === name ? null : name);
+        break;
+    }
+  };
+
+  const hiddenFileInput = React.useRef<any>(null);
+
+  const handleUploadImageClick = () => {
+    if (hiddenFileInput?.current instanceof HTMLInputElement) {
+      hiddenFileInput.current.click();
+    }
+  };
+
+  const handleUpload = async (e: React.ChangeEvent) => {
+    const input = e.target as HTMLInputElement;
+    console.log(home.slogan);
+
+    const inputName = input.name;
+
+    if (!input.files?.length) {
+      return;
+    }
+    const file = input.files[0];
+
+    // display the image
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      setImageLink(e.target?.result as string);
+    };
+    reader.readAsDataURL(file);
+
+    let imageUrl;
+    try {
+      imageUrl = await uploadImage(file);
+      if (imageUrl) {
+        setHome({ ...home, [inputName]: imageUrl });
+      }
+    } catch (e) {
+      console.error("Image size is too large");
+      return;
+    }
+  };
+
+  const handleView = () => {
+    //
+  };
+
   return (
-    <div>EditCoverInfoSection</div>
-  )
-}
+    <section className="contact">
+      <div className="container ml-24 px-8 py-4 md:px-16">
+        <h2 className="mt-12 mb-4 text-xl">home</h2>
+        <h3 className="mb-12">Edit cover information</h3>
+        <div className="mb-4 grid gap-8 md:grid-cols-2 [&_td]:p-2">
+          <div className="col-1">
+            <div className="h-300 relative mb-4 flex">
+              <Image
+                width="500"
+                height="200"
+                src={cover || "/asset1.png"}
+                style={{
+                  objectFit: "cover",
+                }}
+                alt="cover"
+              />
+              <div
+                className="cover-edit-icon"
+                onClick={() => handleOpen("cover")}
+              >
+                <ReactSVG src="/edit.svg" className="edit-svg h-3 w-3" />
+              </div>
+              {isOpen === "cover" && (
```

Comment:
Having a different popup element for each image is very redundant,
We need to use one popup and only change it's location on the screen based on the element